zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Collection.java

```diff
@@ -130,10 +132,28 @@ public CollectionDescription get(HttpServletRequest request, HttpServletResponse
         return result;
     }
 
+    private void requireViewPermission(Session session, String collectionId) throws UnauthorizedException {
+        boolean canView = false;
+        try {
+            canView = permissionsService.canView(session, collectionId);
+        } catch (IOException ex) {
+            throw new UnauthorizedException("You are not authorised to view this collection");
```

Comment:
This exception is thrown if there is an error checking the permission but we are swallowing the original exception. Can you update the message to reflect this, include the user email and collection ID and also include `ex` as a param. Something like:
```
throw new UnauthorizedException(MessageFormat.format("error checking user {0} view permissions for collection {1}", session.getEmail(), collectionID), ex);
```
You may need to add a new constructor to `UnauthorizedException` to take the additional param but I think its important to capture the original error details.

Reply:
I know this code existed before but I'd like to take advantage and improve it while we're here.